src/alpine/gp/primitives.py

```diff
@@ -147,3 +147,53 @@ def add_primitives_to_pset(
                     in_types = primitives_collection[typed_primitive].in_types
                     out_type = primitives_collection[typed_primitive].out_type
                     pset.addPrimitive(op, in_types, out_type, name=typed_primitive)
+
+
+def deap_primitive_to_sympy_expr(prim: Primitive, conversion_rules: Dict, args: Tuple):
+    """Convert a deap scalar primitive into a sympy primitive.
+
+    Args:
+        prim: the primitive.
+        conversion_rules: a dictionary of conversion rules.
+        args: args of the primitive.
+
+    Returns:
+        the converted primitive.
+
+    """
+    prim_formatter = conversion_rules.get(prim.name, prim.format)
+
+    return prim_formatter(*args)
+
+
+def stringify_for_sympy(
+    f: PrimitiveTree, conversion_rules: Dict, special_term_name: str
+) -> str:
+    """Returns a sympy-compatible expression.
+
+    Args:
+        f: the individual tree (DEAP format)
+        conversion_rules: a dictionary of conversion rules.
+        special_term_name: name of the constant placeholder.
+
+    Returns:
+        the sympy-compatible expression.
+    """
+    prim_string = ""
```

Comment:
sorry, I suggested to change "string" with "prim_string" but maybe a better name is "expr"